### ISSUE TITLE
Use nodejs's "path"

### ISSUE THREAD

**Issue by wilhelmberg** [mapbox/kml-split#1]
@mapsam 
you might want to consider using some methods of [`path`](https://nodejs.org/api/path.html) in [here](https://github.com/mapbox/kml-split/blob/master/index.js#L6-L8).

e.g [`path.basename`](https://nodejs.org/api/path.html#path_path_basename_path_ext):

``` js
var sourceName = path.basename(file, '.kml')
```

instead of 

``` js
var paths = file.split('/');
var sourceName = paths[paths.length-1].replace('.kml', '');
```

**Comment by wilhelmberg** [comment]
Also [`path.join`](https://nodejs.org/api/path.html#path_path_join_path) instead of [`out+'/'+name+'-'+count+'.kml'`](https://github.com/mapbox/kml-split/blob/984b39611113e4ff61327126ec440a39fee32ac6/index.js#L73).

**Comment by mapsam** [comment]
ah, nice `path.basename` & `path.join` will be super helpful!

**Comment by mapsam** [comment]
Fixed in 542930596a80fef9166bafe1dab8312b1afa702f

**Comment by wilhelmberg** [comment]
Another one I missed before [`path.dirname()`](https://nodejs.org/api/path.html#path_path_dirname_path):

Use

``` js
path.dirname(file)
```

 instead of

``` js
var paths = file.split('/');
file.replace(paths[paths.length-1], '')
```

No need to fiddle with OS specific directory separators and string arrays.
